runtime/libia2/init.c

```diff
@@ -1,16 +1,25 @@
+#ifndef _GNU_SOURCE
+#define _GNU_SOURCE
+#endif
+#include <dlfcn.h>
+#include <elf.h>
+#include "ia2.h"
 #include "ia2_internal.h"
 #include <sys/auxv.h>
 #include <sys/prctl.h>
 
+/* Pass to mmap to signal end of program init */
+#define IA2_FINISH_INIT_MAGIC 0x1a21face1a21faceULL
```

Comment:
This isn't exactly a "magic", but a random invalid pointer you're passing to `mmap` (it could still collide because of ASLR, but I guess the probability of that is low).